lms/services/assignment.py

```diff
@@ -13,221 +12,64 @@ def __init__(self, db):
 
     @lru_cache(maxsize=128)
     def get(
-        self,
```

Comment:
All of the complexity about having two IDs disappears.  `get` is just a plain query by guid and `resource_link_id`.